jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java

```diff
@@ -505,6 +511,32 @@ public Object execute(org.kie.api.runtime.Context context) {
         	disposeRuntimeEngine(manager, engine);
         }
 	}
+
+	private List<String> getActiveBoundaryNodesSignals(ProcessInstance processInstance, Collection<NodeInstance> activeNodes) {
+        WorkflowProcessInstance workflowProcessInstance = (WorkflowProcessInstance) processInstance;
+        ArrayList<Node> processNodesList = new ArrayList(Arrays.asList(((RuleFlowProcessInstance) workflowProcessInstance).getNodeContainer().getNodes()));
+        List<String> activeBoundaryNodesSignals = new ArrayList<>();
+
+        Map<String, NodeInstance> uniqueIdNodeInstanceMap = new HashMap<>();
+        for(NodeInstance activeNode : activeNodes) {
+            uniqueIdNodeInstanceMap.put((String) activeNode.getNode().getMetaData().get("UniqueId"), activeNode);
+        }
+
+
+        try {
+            for(Node processNode : processNodesList) {
+                if(processNode instanceof BoundaryEventNode) {
+                    activeBoundaryNodesSignals.add(VariableUtil.resolveVariable((String) processNode.getMetaData().get("SignalName"),
+                                                                                uniqueIdNodeInstanceMap.get(((BoundaryEventNode) processNode).getAttachedToNodeId())));
```

Comment:
isn't that part uniqueIdNodeInstanceMap.get can return null and then resolveVariable will throw NPE? Maybe it would be better to avoid that... exceptions might be expensive ...